src/pages/EventPage/EventPage.tsx

```diff
@@ -88,34 +91,48 @@ export default function EventPage({
         };
     };
 }): JSX.Element {
+    const userRecord = useUserRecordContext();
     const [modalShow, setModalShow] = React.useState<boolean>(false);
     const [heatMapData, setHeatMapData] = React.useState<HeatMapData>();
 
     React.useEffect(() => {
-        return getDocSnapshot$(`/events/${match.params.id}`, {
-            next: (snapshot) => {
-                const event = snapshot.data() as EventData;
-                const tempYTimes = getYTimesSorted(event.availability);
-                const tempXDays = getXDaysSorted(
-                    tempYTimes,
-                    event.availability
-                );
-                setHeatMapData({
-                    yData: tempYTimes,
-                    xData: formatXDays(tempXDays),
-                    mapData: createAvailabilityDataArray(
+        if (userRecord) {
+            getDocSnapshot$(`/events/${match.params.id}`, {
+                next: (eventSnapshot) => {
+                    const event = eventSnapshot.data() as EventData;
+                    const tempYTimes = getYTimesSorted(event.availability);
+                    const tempXDays = getXDaysSorted(
                         tempYTimes,
-                        tempXDays,
                         event.availability
-                    ),
-                    zeroState: createZeroStateArray(
-                        tempYTimes.length,
-                        tempXDays.length
-                    ),
-                });
-            },
-        });
-    }, []);
+                    );
+                    const formatedTempXDays = formatXDays(tempXDays);
+                    const zeroState = createZeroStateArray(
+                        LABELS.yLabels.length,
+                        LABELS.xLabels.length
+                    );
+
+                    setHeatMapData({
+                        yData: tempYTimes,
+                        xData: formatedTempXDays,
+                        mapData: createAvailabilityDataArray(
+                            tempYTimes,
+                            tempXDays,
+                            event.availability
+                        ),
+                        preloadData:
+                            Object.values(userRecord.availability).length === 0
+                                ? zeroState
+                                : createPreloadArray(
+                                      tempYTimes,
+                                      formatedTempXDays,
+                                      userRecord.availability
+                                  ),
+                        zeroState,
```

Comment:
Is this still needed? It looks like preloadData will populate a zeroState if no user data is available.

Reply:
The zeroState here could be used in the future, for instance in a clear button.